code/client/src/index.js

```diff
@@ -2,19 +2,32 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+// import { ToastContainer } from 'react-toastify';
+import { ToastContainer } from "../node_modules/react-toastify/dist/react-toastify";
+import { BrowserRouter } from 'react-router-dom';
+import { useState } from "react";
 
-import './scss/index.scss'
-
-import { BrowserRouter } from 'react-router-dom'
-import { ToastContainer } from 'react-toastify'
+import AuthContext from './context/authContext';
+import './scss/index.scss';
+// import '../node_modules/react-toastify/dist/ReactToastify.css';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
+
+const LoginContextApp = () => {
```

Comment:
Instead of writing authentication state here and passing props to App, and recursively to other children components, you may move it in `contexts/AuthContext.jsx` so that here you will able to import from contexts just something like `AuthContext` and wrap everything (as you did) with `AuthContext.Provider`

See also the comment in `AuthContext.jsx`.